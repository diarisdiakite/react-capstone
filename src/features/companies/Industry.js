import React from 'react';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router';
import { Link } from 'react-router-dom';
import {
  selectAllCountries, selectAllCompanies, selectAllSectors,
  selectAllIndustries,
  selectCompaniesByIndustry,
  selectCompaniesByCountry,
} from './companiesSlice';

function Sector() {
  const companies = useSelector(selectAllCompanies);
  console.log(companies); // Ok
  const countries = useSelector(selectAllCountries);
  console.log(countries); // OK
  const sectors = useSelector(selectAllSectors);
  console.log(sectors); // Okay
  const companiesByCountry = useSelector(selectCompaniesByCountry);
  console.log(companiesByCountry);
  const industries = useSelector(selectAllIndustries);
  console.log(industries);

  const { industry } = useParams();
  const companiesByIndustry = useSelector(
    (state) => selectCompaniesByIndustry(state, industry),
  );
  return (
    <div key={industry} className="main-container">
      <h2>{industry}</h2>
      <ul>
        {companiesByIndustry.map((company) => (
          <li key={company.symbol}><Link to={`/companies/${company.symbol}`}>{company.companyName}</Link></li>
        ))}
      </ul>
    </div>
  );
}

export default Sector;
