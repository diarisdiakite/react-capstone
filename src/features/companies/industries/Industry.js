import React from 'react';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router';
import { Link } from 'react-router-dom';
import { selectCompaniesByIndustry } from '../companiesSlice';

function Industry() {
  const { industry } = useParams();
  const companiesByIndustry = useSelector(
    (state) => selectCompaniesByIndustry(state, industry),
  );
  return (
    <div key={industry} className="main-container">
      <h2>{industry}</h2>
      <ul>
        {companiesByIndustry.map((company) => (
          <li key={company.symbol}><Link to={`/companies/${company.symbol}`}>{company.companyName}</Link></li>
        ))}
      </ul>
    </div>
  );
}

export default Industry;
