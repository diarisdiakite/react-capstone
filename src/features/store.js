import { configureStore } from '@reduxjs/toolkit';
import usersReducer from './users/usersSlice';
import companiesReducer from './companies/companiesSlice';

const store = configureStore({
  reducer: {
    users: usersReducer,
    companies: companiesReducer,
  },
});

export default store;
