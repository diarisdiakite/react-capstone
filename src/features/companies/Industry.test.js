import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import store from '../store';
import Industry from './Industry';

describe('RocketsList component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <Industry />
          </MemoryRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
