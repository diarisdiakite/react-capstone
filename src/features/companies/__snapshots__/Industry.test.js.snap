// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketsList component renders correctly 1`] = `
<div
  className="main-container"
>
  <h2 />
  <ul />
</div>
`;
